drivers/shared/src/vic_time.c

```diff
@@ -0,0 +1,794 @@
+        if (jd >= 2299170.5) {
+            // 1582 October 15 (Gregorian Calendar)
+            B = 2 - A + (int)(A / 4);
+        }
+        else if (jd < 2299160.5) {
+            // 1582 October 5 (Julian Calendar)
+            B = 0;
+        }
+        else {
+            log_err("impossible date (falls in gap between end of Julian "
+                    "calendar and beginning of Gregorian calendar");
+        }
+    }
+    else if (calendar == CALENDAR_PROLEPTIC_GREGORIAN) {
+        B = 2 - A + (int)(A / 4);
+    }
+    else if (calendar == CALENDAR_JULIAN) {
+        B = 0;
+    }
+    else {
+        log_err("unknown calendar, must be one of julian,standard,gregorian,"
+                "proleptic_gregoria");
+    }
+
+    // adjust for Julian calendar if necessary
+    jd += B;
+
+    return jd;
+}
+
+/******************************************************************************
+ * @brief   creates a Julian Day for a calendar with no leap years from a
+ *          dmy structure.
+ * @return  Fractional Julian Day.
+ *****************************************************************************/
+double
+_NoLeapDayFromDate(dmy_struct *date)
+{
+    double         jd, day;
+    unsigned short year, month;
+
+    year = date->year;
+    month = date->month;
+
+    // Convert time to fractions of a day
+    day = _FractionalDayFromDate(date);
+
+    // Start Meeus algorithm (variables are in his notation)
+    if (month < 3) {
+        month += MONTHS_PER_YEAR;
+        year -= 1;
+    }
+
+    jd = (double)(unsigned short)(DAYS_PER_YEAR * (year + 4716)) +
+         (double)(unsigned short)(30.6001 * (month + 1)) +
+         day - 1524.5;
+
+    return jd;
+}
+
+/******************************************************************************
+ * @brief   creates a Julian Day for a calendar where all years have 366 days
+ *          from a dmy structure.
+ * @return  Fractional Julian Day.
+ *****************************************************************************/
+double
+_AllLeapFromDate(dmy_struct *date)
+{
+    double         jd, day;
+    unsigned short year, month;
+
+    year = date->year;
+    month = date->month;
+
+    // Convert time to fractions of a day
+    day = _FractionalDayFromDate(date);
+
+    // Start Meeus algorithm (variables are in his notation)
+    if (month < 3) {
+        month += MONTHS_PER_YEAR;
+        year -= 1;
+    }
+
+    jd = (double)(unsigned short)(DAYS_PER_LYEAR * (year + 4716)) +
+         (double)(unsigned short)(30.6001 * (month + 1)) +
+         day - 1524.5;
+
+    return jd;
+}
+
+/******************************************************************************
+ * @brief   creates a Julian Day for a calendar where all months have 30 days
+ *          from a dmy structure.
+ * @return  Fractional Julian Day.
+ *****************************************************************************/
+double
+_360DayFromDate(dmy_struct *date)
```

Comment:
change function name. Once you remove the leading underscore you cannot start a function with a number.